frontend/src/routes/project/[project_code]/+page.svelte

```diff
@@ -8,93 +8,141 @@
 	import t from '$lib/i18n';
 	import type { PageData } from './$types';
 	import DeleteModal from '$lib/components/modals/DeleteModal.svelte';
-	import { _deleteProjectUser, type ProjectUser } from './+page';
+	import {
+		_changeProjectDescription,
+		_changeProjectName,
+		_deleteProjectUser,
+		type ProjectUser,
+	} from './+page';
+	import EditableText from '$lib/components/EditableText.svelte';
+	import { Badge, BadgeList, TaggedBadge } from '$lib/components/Badges';
+	import { z } from 'zod';
+	import { user } from '$lib/user';
 
 	export let data: PageData;
 
 	$: project = data.project;
+	$: _project = project as NonNullable<typeof project>;
 
 	let deleteUserModal: DeleteModal;
-	let userToDelete: ProjectUser|undefined;
+	let userToDelete: ProjectUser | undefined;
 	async function deleteProjectUser(projectUser: ProjectUser) {
 		userToDelete = projectUser;
 		await deleteUserModal.prompt(async () => {
 			await _deleteProjectUser(project!.id, projectUser.User.id);
 		});
 	}
-</script>
 
-{#if project}
-	<div>
-		<p>
-			<span class="text-2xl">{project.name}</span>
-		</p>
+	const canManage = $user?.role == 'admin' || $user?.projects.find(project => project.code == project.code)?.role == 'Manager';
 
-		<div>
-			<span>{project.code}</span>
+	const projectNameValidation = z.string().min(1, $t('project_page.project_name_empty_error'));
+</script>
 
-			<div class="badge badge-lg"><FormatProjectType type={project.type} /></div>
-			<div class="badge badge-lg"><FormatRetentionPolicy policy={project.retentionPolicy} /></div>
+<div class="space-y-4">
+	{#if project}
+		<div class="text-3xl flex items-center gap-3 flex-wrap">
+			<span>Project:</span>
+			<EditableText
+				disabled={!canManage}
+				value={project.name}
```

Comment:
we should probably bind the value here, that when when the editor changes the value it gets updated on the project object, then if anything else is using the name somewhere (like the tab title) it should also get updated.